src/engine/MaxConstraint.cpp

```diff
@@ -492,12 +512,32 @@ bool MaxConstraint::constraintObsolete() const
     return _obsolete;
 }
 
-void MaxConstraint::eliminateVariable( unsigned var, double /*value*/ )
-{
+void MaxConstraint::eliminateVariable( unsigned var, double value )
+{ // todo updated - Guy A
+    if ( !_eliminatedSomeVariables )
+    {
+        _eliminatedSomeVariables = true;
+        _maxValueOfEliminated = value;
+    }
+    else
+    {
+        if ( FloatUtils::gte( value, _maxValueOfEliminated ))
+            _maxValueOfEliminated = value;
+    }
     _elements.erase( var );
-    if ( var == _f || getParticipatingVariables().size() == 1 )
+    if ( var == _f )
         _obsolete = true;
+    if ( getParticipatingVariables().size() == 1 )
+    {
+        unsigned singleVarLeft = *getParticipatingVariables().begin();
+        std::cout<<"inside MAX "<< singleVarLeft<< "\n"; // todo delete
+//        String str; // todo delete
+//        dump(str); // todo delete        if (FloatUtils::gte( getLowerBound( singleVarLeft ), _maxValueOfEliminated ))
+        _obsolete = true;
```

Comment:
I think the constraint is obsolete iff: 
1. you have 1 element left, and no eliminated vars
2. you have 1 element left, and max eliminated value is disjoint from the range of remaining element